src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -562,6 +660,215 @@ class MacroAssembler: public Assembler {
   void andi(Register Rd, Register Rn, int64_t imm, Register tmp = t0);
   void orptr(Address adr, RegisterOrConstant src, Register tmp1 = t0, Register tmp2 = t1);
 
+// Load and Store Instructions
+#define INSN_ENTRY_RELOC(result_type, header)                               \
+  result_type header {                                                      \
+    guarantee(rtype == relocInfo::internal_word_type,                       \
+              "only internal_word_type relocs make sense here");            \
+    relocate(InternalAddress(dest).rspec());
+
+#define INSN(NAME)                                                                                 \
+  void NAME(Register Rd, address dest) {                                                           \
+    assert_cond(dest != NULL);                                                                     \
+    int64_t offset = dest - pc();                                                                  \
+    if (is_offset_in_range(offset, 32)) {                                                          \
+      auipc(Rd, (int32_t)offset + 0x800);                                                          \
+      Assembler::NAME(Rd, Rd, ((int32_t)offset << 20) >> 20);                                      \
+    } else {                                                                                       \
+      int32_t offset = 0;                                                                          \
```

Comment:
It might be confusing that we have already defined `offset` in L673 with a different type.